commit 028402f2d072af2e63acafb000a53af4948491a9
Author: CODeRUS <coderusinbox@gmail.com>
Date:   Mon Feb 2 23:27:08 2015 +0500

    v0.0.2

diff --git a/usr/share/jolla-camera/pages/settings/ExposureMenu.qml b/usr/share/jolla-camera/pages/settings/ExposureMenu.qml
index b668c00..d71896c 100644
--- a/usr/share/jolla-camera/pages/settings/ExposureMenu.qml
+++ b/usr/share/jolla-camera/pages/settings/ExposureMenu.qml
@@ -4,13 +4,17 @@ import Sailfish.Silica 1.0
 import com.jolla.camera 1.0
 
 ExpandingMenu {
-    //% "Exposure compensation"
-    title: qsTrId("jolla-camera-la-exposure_compensation")
-    model: Settings.mode.exposureCompensationValues
+    title: Settings.cameraText
+    model: [ Settings.global["cameraDevice"] ]
     delegate: ExpandingMenuItem {
-        settings: Settings.mode
-        property: "exposureCompensation"
+        settings: Settings.global
+        property: "cameraDevice"
         value: modelData
-        icon: Settings.exposureIcon(modelData)
+        icon: Settings.cameraIcon(modelData)
+    }
+    onClicked: {
+    	open = false
+    	if (Settings.global["cameraDevice"] == "primary" ) Settings.global["cameraDevice"] = "secondary"
+    	else Settings.global["cameraDevice"] = "primary"
     }
 }
diff --git a/usr/share/jolla-camera/pages/settings/SettingsOverlay.qml b/usr/share/jolla-camera/pages/settings/SettingsOverlay.qml
index c9757d4..a2d4f47 100644
--- a/usr/share/jolla-camera/pages/settings/SettingsOverlay.qml
+++ b/usr/share/jolla-camera/pages/settings/SettingsOverlay.qml
@@ -427,14 +427,15 @@ PinchArea {
             SettingsMenu {
                 id: cameraDeviceMenu
                 width: overlay._menuWidth
-                title: Settings.cameraText
+                //% "Exposure compensation"
+                title: qsTrId("jolla-camera-la-exposure_compensation")
                 header: overlay.isPortrait ? lowerHeader : upperHeader
-                model: [ "primary", "secondary" ]
+                model: Settings.mode.exposureCompensationValues
                 delegate: SettingsMenuItem {
-                    settings: Settings.global
-                    property: "cameraDevice"
+                    settings: Settings.mode
+                    property: "exposureCompensation"
                     value: modelData
-                    icon: Settings.cameraIcon(modelData)
+                    icon: Settings.exposureIcon(modelData)
                 }
             }
         }
